### PR TITLE
Fixed up messaging

### PR DESCRIPTION
**Pull request by williamieong**
- Added default picture functionality
- Added progress dialog functionality
- Does not break when there are no dialogs
- Some fixes to the search fragment to get it working

### CODE REVIEW

File: app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java [hunk 1]
```diff
@@ -169,24 +181,24 @@ public void onClick(View view) {
...
         Intent intent = getActivity().getIntent();
-        String city = intent.getStringExtra("city");
-        progress = (SeekBar)view.findViewById(R.id.circleFilter);
-
+        String city = prefs.getString("city", "");
+        progress = (SeekBar) view.findViewById(R.id.circleFilter);
 
 
-        if (city==null) {
+        if (city == "") {
```
Comment (alexrosenfeld10) [inline app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java]: .isEmpty

File: app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java [hunk 2]
```diff
@@ -114,51 +122,55 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                EditText tfLocation = (EditText) view.findViewById(R.id.tfLocation);
+                // String location = tfLocation.getText().toString();
                 //if(! location.equals("")) {
-                    // checks if user entered anything or not "empty string"
-                 //   if (location.equals("Vacuum")) {
-                        lvItems.setVisibility(View.VISIBLE);
-                 //   }
-               // }
+                // checks if user entered anything or not "empty string"
+                //   if (location.equals("Vacuum")) {
```
Comment (alexrosenfeld10) [inline app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java]: this comment worries me

File: app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java [hunk 3]
```diff
@@ -286,30 +305,30 @@ public void onConnected(@Nullable Bundle bundle) {
...
         // if preferences exist
-        // TODO: have it handle if intent is null
-        Intent intent = getActivity().getIntent();
-        Bundle extras = intent.getExtras();
-        if (extras.containsKey("zip")) { // if there is location prefs
-            String postalcode = intent.getStringExtra("zip");
+        if (prefs.contains("zip")) {
+            // if (extras.containsKey("zip")) { // if there is location prefs
+            String postalcode = prefs.getString("zip", "02215");
             String key = "https://maps.googleapis.com/maps/api/geocode/json?address=";
```
Comment (alexrosenfeld10) [inline app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java]: this looks like a URL to me, not a key

File: app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java [hunk 4]
```diff
@@ -453,16 +474,15 @@ public void onStopTrackingTouch(final SeekBar seekBar) {
         this.markers = new HashMap<String, Marker>();
         */
 
+// TODO: RECOMMENT?
```
Comment (alexrosenfeld10) [inline app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java]: not sure what this means

File: app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java (hunk 3 above)
Comment (tkixi) [reply app/src/main/java/com/teamcaffeine/hotswap/navigation/SearchFragment.java]: zip is part of a url (e.g. 02215) that gets appended to the url to get a json object from google maps

**Comment by jschadaway** [comment]
@williamieong I've referenced this PR at https://github.com/TeamCaffeine/HotSwap/issues/46 so that you can patch up some of Alex's points later.

**Pull request merged**